specs/governance/alligator.md

```diff
@@ -0,0 +1,207 @@
+# Alligator
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Interface](#interface)
+  - [Core Functions](#core-functions)
+    - [`subdelegate`](#subdelegate)
+    - [`subdelegateFromToken`](#subdelegatefromtoken)
+    - [`subdelegateBatched`](#subdelegatebatched)
+    - [`afterTokenTransfer`](#aftertokentransfer)
+  - [Getters](#getters)
+    - [`getSubdelegations`](#getsubdelegations)
+    - [`getCheckpoints`](#getcheckpoints)
+    - [`getVotingPower`](#getvotingpower)
+- [Storage](#storage)
+- [Types](#types)
+  - [`SubdelegationRule`](#subdelegationrule)
+  - [`AllowanceType`](#allowancetype)
+  - [`Checkpoint`](#checkpoint)
+- [Backwards Compatibility](#backwards-compatibility)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations for the [`GovernanceToken`](gov-token.md). Subdelegations enable
```

Comment:
Are any of these advanced delegation use cases documented anywhere? It would be really helpful if they were documented in this spec

Reply:
added a section with the user flow and usecases.